src/FSharpVSPowerTools.Core/SourceCodeClassifier.fs

```diff
@@ -88,11 +88,11 @@ module private StringCategorizers =
             [r.StartLine..r.EndLine]
             |> List.map (fun line ->
                 let lineStr = getTextLine (line - 1)
-                if lineStr.Length > r.StartColumn + 1 && line = r.StartLine && line = r.EndLine && r.StartColumn + 1 <= r.EndColumn - 1 then
+                if line = r.StartLine && line = r.EndLine && r.StartColumn + 1 <= r.EndColumn - 1 && lineStr.Length > r.EndColumn - 1 then
                     lineStr.[r.StartColumn + 1 .. r.EndColumn - 1], line, r.StartColumn + 1
-                elif lineStr.Length > r.StartColumn + 1 && line = r.StartLine then 
+                elif line = r.StartLine && lineStr.Length > r.StartColumn + 1 then 
                     lineStr.[r.StartColumn + 1 ..], line, r.StartColumn + 1
-                elif lineStr.Length > r.EndColumn - 1 && line = r.EndLine then
+                elif line = r.EndLine && lineStr.Length > r.EndColumn - 1 then
                     lineStr.[..r.EndColumn - 1], line, 0
                 else lineStr, line, 0)
 
```

Comment:
I don't think it helps much. I particularly hate the partial nature of those indexers. If only they return option or choice values...